extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -709,7 +709,7 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 
 	getUntitledNotebookUri(resource: string): vscode.Uri {
 		let untitledFileName = vscode.Uri.parse(`untitled:${resource}`);
-		if (!this.currentBook.getAllNotebooks().get(untitledFileName.fsPath) && !this.currentBook.getAllNotebooks().get(path.basename(untitledFileName.fsPath))) {
+		if (this.currentBook && !this.currentBook.getAllNotebooks().get(untitledFileName.fsPath) && !this.currentBook.getAllNotebooks().get(path.basename(untitledFileName.fsPath))) {
```

Comment:
What is this even doing? A function named `get...` I wouldn't expect to have side effects like setting a path to notebook mapping.

And why are we calling basename on an untitled path anyways?

@barbaravaldez Are you familiar with this at all?

Reply:
Completely agree on questions re: why this function is actually adding something to a map, given its name, and I certainly don't have answers for why we're checking basename on an untitled path.

Reply:
@Charles-Gagnon I'm not familiar with this bit of code but I agree that it doesn't make sense

Reply:
@chlafreniere Can you : 

1. Update the type of currentbook (`public currentBook: BookModel | undefined;`)
2. Look through the rest of the code and make sure it's being handled correctly elsewhere - I see a couple of places that don't do a null check